benchmarks/compression.py

```diff
@@ -23,6 +24,12 @@ def open_write(self, path: str) -> IO[bytes]:
     def open_read(self, path: str) -> IO[bytes]:
         raise NotImplementedError
 
+    def write_stream(self, fobj: IO[bytes]) -> IO[bytes]:
```

Comment:
This is a good refactor by the way to make the API more general. Great job! In order to enable this pull request to be more targeted and to ensure that all the tests also get updated, I've pull out this refactor into a separate PR: https://github.com/ascoderu/compression-benchmarks/pull/3

Reply:
The pull request is now merged. Please rebase off of master to get the changes. Thanks!